plugins/azure/resoto_plugin_azure/azure_client.py

```diff
@@ -66,8 +70,16 @@ def list(self, spec: AzureApiSpec, **kwargs: Any) -> List[Json]:
             else:
                 raise e
 
-    def delete(self, resource_id: str) -> None:
-        self.client.resources.delete_by_id(resource_id)
+    def delete(self, resource_id: str) -> bool:
+        try:
+            self.client.resources._delete_by_id_initial(resource_id=resource_id, api_version="2021-04-01")
```

Comment:
`_delete_by_id_initial` the underscore marks it as internal API.
I am willing to use it - I want to ensure it is required.
Is there a good reason to not use `delete_by_id`?